external/llvm-project/mlir/lib/Conversion/AsyncToGPU/AsyncToGPU.cpp

```diff
@@ -230,9 +218,13 @@ class LaunchOpConversion : public OpConversionPattern<async::LaunchOp> {
       auto fidx = i - diff;
       Value opr = operands[i];
       // move input memories to GPU
-      if (!opr.getDefiningOp<gpu::AllocOp>()) {
-        opr =
-            moveMemory(rw, opr, fidx, accessVec[fidx], copyBackOprs, asyncDeps);
+      if (opr.getType().isa<MemRefType>() &&
+          !opr.getDefiningOp<gpu::AllocOp>()) {
+        bool readAccess{func.getArgAttr(fidx, FuncOp::getReadAccessAttrName())};
```

Comment:
Why are these two otherwise-identical operations done in two different ways?

Reply:
Forgot to update the second. I wanted to get rid of the C-style cast.